File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode DeleteDuplicates(ListNode head) {
        ListNode cur = head;
        while (cur != null) {
            if (cur.next != null && cur.val == cur.next.val) {
                ListNode next = cur.next;
                while (next.next != null && next.val == next.next.val) {
                    next = next.next;
                }
                cur.next = next.next;
            }
            cur = cur.next;
        }
        return head;
    }
}

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
               ListNode* cur = head;
        while(cur) {
            if(cur->next and cur->val == cur->next->val) {
                ListNode* next = cur->next;
                while(next->next and next->val == next->next->val) next = next->next;
                cur->next = next->next;
            }
            cur = cur->next; 
        }


        return head;
    }
};